examples/AlgorandDBStorage/AlgorandDBStorage.algo.ts

```diff
@@ -0,0 +1,66 @@
+import { Contract } from "@algorandfoundation/tealscript";
+
+// eslint-disable-next-line no-unused-vars
+class AlgorandDBStorage extends Contract {
+  data = new BoxMap<bytes, bytes>();
+  /**
+   * Add key to the algorand DB Storage.
+   * If key exists, tx will fail.
+   *
+   * @param key DB key
+   * @param value DB Value
+   */
+  add(key: string, value: bytes): void {
+    assert(this.txn.sender === globals.creatorAddress);
+    assert(!this.data.exists(key));
+    this.data.set(key, value);
+  }
+  /**
+   * Update the value at the key at the algorand DB Storage.
+   * If key does not exists, tx will fail.
+   *
+   * @param key DB key
+   * @param value DB Value
+   */
+  update(key: string, value: bytes): void {
+    assert(this.txn.sender === globals.creatorAddress);
+    assert(this.data.exists(key));
+    this.data.delete(key);
+    this.data.set(key, value);
+  }
+  /**
+   * Update the value at the key at the algorand DB Storage.
+   * If key does not exists, tx will not fail, but will add the item to the storage.
+   *
+   * @param key DB key
+   * @param value DB Value
+   */
+  upsert(key: string, value: bytes): void {
+    assert(this.txn.sender === globals.creatorAddress);
+    this.data.set(key, value);
+  }
+  /**
+   * Deletes the key at the algorand DB Storage.
+   * If key does not exists, tx will fail.
+   *
+   * @param key DB key
+   */
+  delete(key: string): void {
+    assert(this.txn.sender === globals.creatorAddress);
+    assert(this.data.exists(key));
+    this.data.delete(key);
+  }
+
+  // /**
```

Comment:
TEALScript now supports booleans, so you can uncomment this method if you'd like. Auto-generated `supportsInterface` is on the TODO list.